### PR TITLE
Cryp 73 implement frontend component of a transaction

### PR DESCRIPTION
**Pull request by alexbara2000**
Added Transaction component

### CODE REVIEW

File: packages/client/components/Transactions.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    },
+    transactionAmountOut: {
+        marginLeft: "auto",
+        fontSize: 17,
+        fontWeight: "600",
+    },
+    transactionAmountIn: {
+        marginLeft: "auto",
+        fontSize: 17,
+        color: "#16A34A",
```
Comment (domsec) [inline packages/client/components/Transactions.tsx:93]: Could you move the color as a property in the associated component so color identification is easier. The only time we should be adding the color hex value is for icons.

File: packages/client/components/Transactions.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        fontWeight: "600",
+    },
+    transactionDate: {
+        fontSize: 15,
+        color: "#737373",
```
Comment (domsec) [inline packages/client/components/Transactions.tsx:98]: Same as above

File: packages/client/components/Transactions.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    },
+    transactionCurrency: {
+        fontSize: 15,
+        color: "#737373",
```
Comment (domsec) [inline packages/client/components/Transactions.tsx:102]: Same as above

File: packages/client/components/Transactions.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                    </HStack>
+                </VStack>
+            </HStack>
+        </Box>
+    );
+}
+
+const styles = StyleSheet.create({
+    sendIcon: {
+        color: "#404040",
```
Comment (domsec) [inline packages/client/components/Transactions.tsx:66]: Add the color name as a comment to better identify the color.

File: packages/client/components/Transactions.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        marginLeft: 0,
+        marginRight: 10,
+        alignSelf: "center",
+    },
+    receiveIcon: {
+        color: "#16A34A",
```
Comment (domsec) [inline packages/client/components/Transactions.tsx:72]: Same as above comment

**Review by domsec — APPROVED** [review]
Besides the suggested refactors, everything looks good.

File: packages/client/components/Transactions.tsx [hunk 6]
```diff
@@ -46,14 +46,14 @@ export function FormatTransaction({ transactions, wallet }: Props) {
                         <Text style={styles.transactionsAddress}>
                             {formatTransactionAddress(transactions.transactionAddress)}
                         </Text>
-                        <Text style={isIncommingTransaction ? styles.transactionAmountIn : styles.transactionAmountOut}>
+                        <Text color={isIncommingTransaction? "success.600" : "text.600"} style={isIncommingTransaction ? styles.transactionAmountIn : styles.transactionAmountOut}>
```
Comment (domsec) [inline packages/client/components/Transactions.tsx:49]: Should be `text.700` not `text.600`

**Review by domsec — APPROVED** [review]

File: packages/client/components/Transactions.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { Text, HStack, Box, VStack } from "native-base";
+import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
+import { StyleSheet } from "react-native";
+import { Transaction } from "@cryptify/common/src/domain/entities/transaction";
+import { faCircleArrowDownLeftCustom } from "./icons/faCircleArrowDownLeftCustom";
+import { faCircleArrowUpRightCustom } from "./icons/faCircleArrowUpRightCustom";
+
+type Props = {
+    transactions: Transaction;
```
Comment (jason-gerard) [inline packages/client/components/Transactions.tsx:10]: In english when there is an 's' at the end of a word it mean plural but here we only deal with a single 'transaction' so there wouldn't be an 's'. It is singular.

```suggestion
    transaction: Transaction;
```

File: packages/client/components/Transactions.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+    function formatTransactionAddress(address: string): string {
+        return `${address.substring(0, 6)}...${address.substring(address.length - 4, address.length)}`;
+    }
+
+    function getCurrencyType(): string {
+        return transactions.transactionAddress.substring(0, 2) == "0x" ? "ETH" : "BTC";
+    }
+
+    function getFromattedDate(date: Date): string {
```
Comment (jason-gerard) [inline packages/client/components/Transactions.tsx:25]:
```suggestion
    function getFormattedDate(date: Date): string {
```

File: packages/client/components/Transactions.tsx [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+import { Text, HStack, Box, VStack } from "native-base";
+import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
+import { StyleSheet } from "react-native";
+import { Transaction } from "@cryptify/common/src/domain/entities/transaction";
+import { faCircleArrowDownLeftCustom } from "./icons/faCircleArrowDownLeftCustom";
+import { faCircleArrowUpRightCustom } from "./icons/faCircleArrowUpRightCustom";
+
+type Props = {
+    transactions: Transaction;
+    wallet: string;
```
Comment (jason-gerard) [inline packages/client/components/Transactions.tsx:11]: address or walletAddress here

```suggestion
    walletAddress: string;
```

File: packages/client/components/Transactions.tsx [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+};
+
+export function FormatTransaction({ transactions, wallet }: Props) {
```
Comment (jason-gerard) [inline packages/client/components/Transactions.tsx:14]: Name the file and the component the same thing

```suggestion
export function Transaction({ transactions, wallet }: Props) {
```

**Review by jason-gerard — APPROVED** [review]
Good job, please add screenshots

**Pull request merged**